Library/brew.rb

```diff
@@ -97,7 +97,7 @@ def require?(path)
 
   cmd = HOMEBREW_INTERNAL_COMMAND_ALIASES.fetch(cmd, cmd)
 
-  sudo_check = %w[ install link pin unpin upgrade ]
+  sudo_check = %w[ install reinstall postinstall link pin unpin upgrade ]
```

Comment:
If nobody objects by the time I start to tread towards sleep late tonight I'll commit this with the extra suggested commands.